app/script/cryptography/CryptographyRepository.js

```diff
@@ -355,36 +332,71 @@ z.cryptography.CryptographyRepository = class CryptographyRepository {
       });
   }
 
-  _encryptGenericMessageForNewSessions(recipientsWithMissingSessions, genericMessage) {
-    if (Object.keys(recipientsWithMissingSessions).length) {
-      return this.getUsersPreKeys(recipientsWithMissingSessions)
-        .then(userPreKeyMap => {
-          this.logger.info(`Fetched pre-keys for '${Object.keys(userPreKeyMap).length}' users.`, userPreKeyMap);
+  _encryptGenericMessage(recipients, genericMessage, messagePayload) {
+    return Promise.resolve()
+      .then(() => {
+        const cipherPayloadPromises = [];
+
+        for (const userId in recipients) {
+          const clientIds = recipients[userId];
 
-          const newSessionPromises = [];
+          messagePayload.recipients[userId] = messagePayload.recipients[userId] || {};
+          clientIds.forEach(clientId => {
+            const sessionId = this._constructSessionId(userId, clientId);
+            cipherPayloadPromises.push(this._encryptPayloadForSession(sessionId, genericMessage));
+          });
+        }
 
-          for (const userId in userPreKeyMap) {
-            const clientPreKeyMap = userPreKeyMap[userId];
+        const receivingUsers = Object.keys(messagePayload.recipients).length;
+        const logMessage = `Encrypting message of type '${genericMessage.content}' for '${receivingUsers}' users.`;
```

Comment:
We shoudl have the same log messages also for `_encryptGenericMessageForMissingRecipients`.